pkg/http/api.go

```diff
@@ -1367,123 +1372,19 @@ func (a *api) onDeleteState(reqCtx *fasthttp.RequestCtx) {
 	respond(reqCtx, withEmpty())
 }
 
-func (a *api) onGetSecret(reqCtx *fasthttp.RequestCtx) {
-	store, secretStoreName, err := a.getSecretStoreWithRequestValidation(reqCtx)
-	if err != nil {
-		log.Debug(err)
-		return
-	}
-
-	metadata := getMetadataFromRequest(reqCtx)
-
-	key := reqCtx.UserValue(secretNameParam).(string)
-
-	if !a.isSecretAllowed(secretStoreName, key) {
-		msg := NewErrorResponse("ERR_PERMISSION_DENIED", fmt.Sprintf(messages.ErrPermissionDenied, key, secretStoreName))
-		respond(reqCtx, withError(fasthttp.StatusForbidden, msg))
-		return
-	}
-
-	req := secretstores.GetSecretRequest{
-		Name:     key,
-		Metadata: metadata,
-	}
-
-	start := time.Now()
-	policyRunner := resiliency.NewRunner[*secretstores.GetSecretResponse](reqCtx,
-		a.resiliency.ComponentOutboundPolicy(secretStoreName, resiliency.Secretstore),
-	)
-	resp, err := policyRunner(func(ctx context.Context) (*secretstores.GetSecretResponse, error) {
-		rResp, rErr := store.GetSecret(ctx, req)
-		return &rResp, rErr
+func (a *api) onGetSecretHandler() fasthttp.RequestHandler {
+	return UniversalFastHTTPHandler(a.universal.GetSecret, func(reqCtx *fasthttp.RequestCtx, in *runtimev1pb.GetSecretRequest) {
+		in.StoreName = reqCtx.UserValue(secretStoreNameParam).(string)
```

Comment:
This is the (optional) `inModifier` handler. Its input is the proto object decoded from the request body (if any). In order for HTTP APIs to remain RESTful, the `inModifier` allows changing the proto object that was decoded to add properties read from the URL (for example)